# 1. Find out how many males and females participated in the test.


In [1]:
import numpy as np
import pandas as pd

In [2]:
data=pd.read_csv("StudentsPerformance.csv")

In [3]:
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [22]:
group=data.groupby('gender')

In [23]:
group.count()

race/ethnicity  parental level of education  lunch  \
gender                                                       
female             518                          518    518   
male               482                          482    482   

        test preparation course  math score  reading score  writing score  
gender                                                                     
female                      518         518            518            518  
male                        482         482            482            482

There are 518 Females, and 482 Males

# 2. What do you think about the students' parental level of education?

In [25]:
groupg=data.groupby('parental level of education')

In [28]:
groupg.count()

gender  race/ethnicity  lunch  \
parental level of education                                  
associate's degree              222             222    222   
bachelor's degree               118             118    118   
high school                     196             196    196   
master's degree                  59              59     59   
some college                    226             226    226   
some high school                179             179    179   

                             test preparation course  math score  \
parental level of education                                        
associate's degree                               222         222   
bachelor's degree                                118         118   
high school                                      196         196   
master's degree                                   59          59   
some college                                     226         226   
some high school                                 179         179   

                             reading score  writing score  
parental level of education                                
associate's degree                     222            222  
bachelor's degree                      118            118  
high school                            196            196  
master's degree                         59             59  
some college                           226            226  
some high school                       179            179

The students parental level of education is highest in "some college", lowest in "master's degree"

# 3. Who scores the most on average for math, reading and writing based on
● Gender
● Test preparation course


In [58]:
group3=data.groupby('gender')

In [60]:
group3.mean()

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

In [61]:
group4=data.groupby('test preparation course')

In [62]:
group4.mean()

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

# 4. What do you think about the scoring variation for math, reading and writing based on
● Gender
● Test preparation course


In [67]:
group5=data.groupby('gender')

In [68]:
group5.var()

math score  reading score  writing score
gender                                          
female  256.958593     207.677438     220.369327
male    211.889097     200.211010     202.413924

In [69]:
group6=data.groupby('test preparation course')

In [70]:
group6.var()

math score  reading score  writing score
test preparation course                                          
completed                210.884027     183.265864     175.202612
none                     246.668662     213.419851     226.251739

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this.


In [84]:
data.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000